Various fixes, folder redesign

Flaksim opened the pull request:
changes: 
  - maptweak: "Fixed gravity and power issues on Geminus (Someone forgot areas when mapping new things in...)"
  - rscdel: "Cleaned up map folders to get rid of outdated and horribly broken maps."
  - bugfix: "Implemented XO-11's movement fix."
  - tweak: "Rounded out items here and there, nothing major, this update is mostly fixing little things and step 1 in src reorganization."

Flaksim committed:
Folder cleanup

Rearranged folders and files.

Flaksim committed:
Various changes see changelog

Changelog for info.

Flaksim commented:
The maps fail because the dmb is setup for FC mode, not for geminus, files it needs for those maps arent included in that load order, its not because something is wrong with them.

bloxgate commented:
You'll need to fix that. Including the map/gamemode's code file should include everything the map needs to run, and not stuff from other modes/maps.

Edit: It's also complaining about a missing map for geminus city

Flaksim commented:
Yeah, that map is actually gone, removed it from the includes file now.

As for including everything. Its either loading a shitton of extra files, or a couple of months rebuilding the shit pile people who came before made of it, too many items that are needed over different maps and gamemodes were put in specific subfolders or made dependent on files exclusive to another map/gamemode.

bloxgate commented:
Personally, I'd prefer if the code we have on alpha passes all Travis test. Having people get used to it failing will eventually cause an actual problem to get ignored at some point.

We may have to just bit the bullet and fix our horrible include situation. I'd put that more on the order of a few hours of serious work, rather than a month though.

Flaksim committed:
Travis appeasement

Once more!

Flaksim committed:
Includes fiddling!

Flaksim committed:
travis wont be happy still tho.

Flaksim committed:
in progress, commit to switch work location

Flaksim committed:
Finishes off FC mode.

This commit makes it so FC mode only requires the "all includes" file in the first contact folder to be included in the load order.
It adds proper area mapping and a cleanup of includes files.

Flaksim committed:
Some fixes travis should like.

Flaksim committed:
and another one, just fiddling to get the includes right now on the tests.

erty80s commented on maps/first_contact/maps/UNSC_Corvette.dmm:
```diff
@@ -1,438 +1,462 @@
+"aj" = (/obj/machinery/light{dir = 1},/turf/simulated/floor/plating,/area/om_ships/unscpatrol/oxystorage)
+"ak" = (/obj/machinery/shower{dir = 4; icon_state = "shower"; pixel_x = 5},/obj/machinery/light{tag = "icon-bulb0 (WEST)"; icon_state = "bulb0"; dir = 8},/turf/simulated/floor/tiled/freezer,/area/om_ships/unscpatrol/captainroom)
+"al" = (/obj/structure/curtain/open,/turf/simulated/floor/tiled/freezer,/area/om_ships/unscpatrol/captainroom)
+"am" = (/obj/structure/sink{dir = 1; pixel_y = 16},/obj/structure/mirror{pixel_x = 0; pixel_y = 30},/turf/simulated/floor/tiled/freezer,/area/om_ships/unscpatrol/captainroom)
```
This pixel_x and pixel_y might be why it doesnt pass the travis check. Correct me if I'm wrong

Flaksim replied:
pixel_x and Y are intended to be used, what you CANNOT use is step_x and y

Flaksim committed:
still working on it.

Flaksim committed:
example map test fix attempt 2

Flaksim committed:
another try

testing

Flaksim committed:
test hud fix

Flaksim committed:
hud icons fix

Flaksim committed:
In progress.

Flaksim committed:
lobby fix

Flaksim committed:
another try.

Flaksim committed:
typo fix

Flaksim committed:
This should conclude the reorganization and make travis happy.

Flaksim committed:
This should do the trick.

Flaksim committed:
another include missed.

bloxgate commented on maps/example/example_define.dm:
```diff
@@ -3,14 +3,14 @@
 
-	allowed_spawns = list("Arrivals Shuttle")
+	allowed_jobs = list(/datum/job/example)
+	allowed_spawns = list("Test Spawn")
```
Reason for changing this?

Flaksim replied:
turned it into a localised test contained to the examples file
the way it did spawns on the file it was using to test wasnt even the same as the way we do spawns, so it was a pretty useless test for our purposes.

bloxgate reviewed (commented):
Yeah looks good. Just have one small question about the change to the example map though.

Desolane reviewed (approved):
I give it a semi golden ticket of approval, albeit the light for the helmet buff irks me and the area code has some redundancy, otherwise, keep up the trend of adding armor thickness to helmets and we'll see how it effects gameplay before opting to remove it.

Flaksim committed:
Lays the groundwork for ODST Helmet modifications.

*For internal use: odst.dmi now contains transparent helmet icon states. to be coded in.

Flaksim committed:
little thing

Flaksim committed:
Fixes the last few gamemode include files.

caelaislinn commented on code/game/objects/items/weapons/tanks/tank_types.dm:
```diff
@@ -131,6 +131,13 @@
+	volume = 5
+	canremove = 0
+	icon_state = "emergency_engi"
+	item_state = null
```
I'd like to keep halo items in the modules folder for preference

Flaksim replied:
This is in progress, its being worked on, but It will be removed from that file in the next PR.

caelaislinn commented on code/modules/clothing/suits/storage.dm:
```diff
@@ -57,5 +57,3 @@
 /obj/item/clothing/suit/storage/vest/tactical/New()
 	..()
 	pockets = new/obj/item/weapon/storage/internal/pockets(src, slots = 4, slot_size = 2)
```
Unnecessary line changes

Flaksim replied:
Indeed, its because something was tried before, and I undid it again.

caelaislinn commented on code/modules/halo/clothing/odst.dm:
```diff
@@ -1,5 +1,7 @@
 #define ODST_OVERRIDE 'code/modules/halo/clothing/odst.dmi'
 #define ITEM_INHAND 'code/modules/halo/clothing/odst_items.dmi'
+/obj/item/clothing/suit/armor/odst/New()
+	..()
```
This is redundant

Flaksim replied:
A leftover from an attempt to make integrated oxytanks, the next pr i'm working on deals mainly with that file and removes redundant stuff like that.

caelaislinn commented on code/modules/halo/structures/diesel_gen.dm:
```diff
@@ -1,90 +0,0 @@
-
-/obj/machinery/diesel_generator/New()
-	..()
-	name = name + " #[rand(0,9)]"
```
This would be better done as a static var  counting the number of instances

Flaksim replied:
Diesel gens are gone. the file was removed.

caelaislinn replied:
Just a thought for if they come back.

caelaislinn commented on code/modules/halo/structures/diesel_gen.dm:
```diff
@@ -1,90 +0,0 @@
-	icon_state = "phoron"
-	gauge_icon = null
-	flags = CONDUCT
-	slot_flags = null	//they have no straps!
```
Why was this entire file cleared? I can't see if its re-added

Flaksim replied:
Diesel gens are gone. the file was removed. Its unused and I was asked to clear out some stuff we wouldnt need.

caelaislinn replied:
Our only problem is with having too many zlevels and map files. This kind of content isn't a problem at all.

Flaksim replied:
The mapfiles will be shrunk in the next PR aswell, there are several that can go imo, but for insurrection for example, the new map isnt fully compatible with the mode yet, I'm in the process of doing that, then the old ones can go. All those maps will be moved to a "map archive" folder, and just unused untill/IF we want to revisit them.

As for all the Z levels and mapfiles being used: FC and Geminus modes individually dont have too many Z levels, but we've been running on a "mode" that just throws all the FC maps into Geminus, which is a bad idea.
Will take that up in staff to see what we will and wont include in Geminus.

caelaislinn replied:
My point was that code like this can stay around until we start using it.

caelaislinn reviewed (changes requested):
Requesting some tweaks and clarifications

Flaksim committed:
DONT MERGE IN PROGRESS AGAIN

Flaksim committed:
test

test

Flaksim committed:
Merge branch 'alpha' of https://github.com/Flaksim/HaloSpaceStation13 into alpha

Flaksim committed:
Cut the work in progress out of used code so its mergable.

Merge this, finishing up on that internals code will be a next PR.

Flaksim committed:
Sacrifice to travis.

Flaksim committed:
Fixes path.

Flaksim commented:
Its done for the moment, merge this, makes the game playable again ;)
